addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -309,6 +305,20 @@ var LinkDialog = Dialog.extend({
         const el = this.el.querySelector('[name="link_style_color"]:checked');
         $(this.buttonOptsCollapseEl).collapse(el && el.value ? 'show' : 'hide');
     },
+    /**
+     * Update UI depending on target node.
+     *
+     * @private
+     */
+    _initDialogUI(handleAsButton) {
+        if (handleAsButton) {
+            this.$('input[name="is_new_window"]').closest('.form-group').remove();
+            this.$('input[name="url"]').closest('.form-group').addClass('d-none');
+            if (this.isButton) {
+                this.$("#link-preview").replaceWith($('<button id="link-preview"/>'));
+            }
+        }
+    },
```

Comment:
See previous comment: should thus be in qweb, the diff of this second commit would be 1 line.